shared/src/business/entities/docketEntry/DocketEntryFactory.test.js

```diff
@@ -256,4 +256,66 @@ describe('DocketEntryFactory', () => {
       expect(errors()).toBeNull();
     });
   });
+
+  describe('partyPrimary', () => {
+    beforeEach(() => {
+      rawEntity.dateReceived = createISODateString();
+      rawEntity.documentType = 'Notice of Change of Address';
+      rawEntity.eventCode = 'NCA';
+    });
+
+    it('should not be required when isAutoGenerated is true', () => {
+      rawEntity.isAutoGenerated = true;
+      rawEntity.hasOtherFilingParty = false;
+      rawEntity.otherFilingParty = undefined;
+      rawEntity.partyPrimary = true;
+
+      expect(errors()).toBeNull();
+    });
+
+    it('should be required when isAutoGenerated is false', () => {
+      rawEntity.isAutoGenerated = false;
+
+      expect(errors()).toEqual({
+        partyPrimary: 'Select a filing party',
+      });
+    });
+
+    it('should be not required when isAutoGenerated is true and partyPrimary is true', () => {
+      rawEntity.isAutoGenerated = false;
+      rawEntity.partyPrimary = true;
+
+      expect(errors()).toBeNull();
+    });
+
+    it('should be not required when isAutoGenerated is true, partyPrimary is false, partySecondary is true', () => {
```

Comment:
Same issue as above, `isAutoGenerated` is set to `false` below, even though this description says it is 'true'.